src/sentry/buffer/redis.py

```diff
@@ -288,6 +293,29 @@ def get_hash(
         key = self._make_key(model, field)
         return self._execute_redis_operation(key, RedisOperation.HASH_GET_ALL)
 
+    def handle_pending_partitions(self, partition: int | None) -> None:
```

Comment:
might be nice to have a test for this new function as well

Does this code also need to be DRY'd up in `incr`?